### PR TITLE
issue: fix timestamp updating

### PR DESCRIPTION
**Pull request by Kosadchiy**
Pull request with fix issue timestamp updating [#6209](https://github.com/gogs/gogs/issues/6209).
As you are using xorm events like BeforeUpdate () together with Cols (cols ...) method, it changes only fields wich you pass to Cols () method and BeforeUpdate () doesn’t work.
I think you also should check pull, release and repo updating. It may contains the same problem.

### CODE REVIEW

**Comment by manfer** [comment]
I have not tested but I wonder if you can just solve this with:

```go
func updateIssueCols(e Engine, issue *Issue, cols ...string) error {
        cols = append(cols, "deadline_unix", "updated_unix")
	_, err := e.ID(issue.ID).Cols(cols...).Update(issue)
	return err
}
```

File: internal/db/issue.go [hunk 1]
```diff
@@ -407,14 +407,14 @@ func (issue *Issue) ReadBy(uid int64) error {
 	return UpdateIssueUserByRead(uid, issue.ID)
 }
 
-func updateIssueCols(e Engine, issue *Issue, cols ...string) error {
-	_, err := e.ID(issue.ID).Cols(cols...).Update(issue)
+func updateIssueCols(e Engine, issue *Issue) error {
```
Comment (unknwon) [inline internal/db/issue.go:410]: I think the function should be renamed as well
```suggestion
func updateIssue(e Engine, issue *Issue) error {
```

File: internal/db/issue.go [hunk 2]
```diff
@@ -407,14 +407,14 @@ func (issue *Issue) ReadBy(uid int64) error {
...
+	_, err := e.ID(issue.ID).Update(issue)
 	return err
 }
 
 // UpdateIssueCols only updates values of specific columns for given issue.
-func UpdateIssueCols(issue *Issue, cols ...string) error {
-	return updateIssueCols(x, issue, cols...)
+func UpdateIssueCols(issue *Issue) error {
```
Comment (unknwon) [inline internal/db/issue.go:416]:
```suggestion
func UpdateIssue(issue *Issue) error {
```

**Comment by Kosadchiy** [comment]
> I have not tested but I wonder if you can just solve this with:
> 
> ```go
> func updateIssueCols(e Engine, issue *Issue, cols ...string) error {
>         cols = append(cols, "deadline_unix", "updated_unix")
> 	_, err := e.ID(issue.ID).Cols(cols...).Update(issue)
> 	return err
> }
> ```

I chosed @manfer decision. Obviosly it is better way. I have tested it works

**Review by unknwon — APPROVED** [review]
Thank you!

**Comment by unknwon** [comment]
This patch has been cut with 0.12.1 release.

**Pull request merged**